### PR TITLE
Make mypy stricter

### PR DESCRIPTION
**Pull request by smessmer**
- Make mypy able to find and follow the imports to caffe2 and onnx
- Enable more mypy checks
- Fix warnings caused by these changes

Please ignore failing CI, it's not set up correctly yet.

### CODE REVIEW

File: stubs/numpy/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,498 @@
+"""
+Numpy's mypy stub. Only type declarations for ndarray, the scalar hierarchy and array creation
+methods are provided.
+"""
+
+# This file is taken from https://github.com/machinalis/mypy-data/tree/master/numpy-mypy/numpy
```
Comment (smessmer) [inline stubs/numpy/__init__.pyi]: Question: Can I use this code or are there licensing issues?

Comment (smessmer) [reply stubs/numpy/__init__.pyi]: cc @dzhulgakov for licensing issues

Comment (dzhulgakov) [reply stubs/numpy/__init__.pyi]: BSD and MIT are very close. As long as we maintain reference to the source it's fine. So it's ok to land it

**Comment by dzhulgakov** [comment]
Also - travis doesn't seem to be setup correctly yet.

**Comment by smessmer** [comment]
Thanks. Travis support isn't landed yet.



Sent from Samsung Mobile on O2


-------- Original message --------

**Pull request merged**